addons/account/models/account_move.py

```diff
@@ -1674,34 +1674,31 @@ def _check_balanced(self, container):
         # It happens as the ORM makes the create with the 'no_recompute' statement.
-        self.env['account.move.line'].flush_model(['balance', 'currency_id', 'move_id'])
+        self.env['account.move.line'].flush_model(['balance', 'credit', 'debit', 'currency_id', 'move_id'])
         self.env['account.move'].flush_model(['journal_id'])
         self._cr.execute('''
-            WITH error_moves AS (
-                SELECT line.move_id,
-                       ROUND(SUM(line.balance), currency.decimal_places) balance
-                  FROM account_move_line line
-                  JOIN account_move move ON move.id = line.move_id
-                  JOIN account_journal journal ON journal.id = move.journal_id
-                  JOIN res_company company ON company.id = journal.company_id
-                  JOIN res_currency currency ON currency.id = company.currency_id
-                 WHERE line.move_id IN %s
-              GROUP BY line.move_id, currency.decimal_places
-            )
-            SELECT *
-              FROM error_moves
-             WHERE balance !=0
+            SELECT line.move_id,
+                   ROUND(SUM(line.debit), currency.decimal_places) debit,
+                   ROUND(SUM(line.credit), currency.decimal_places) credit
+              FROM account_move_line line
+              JOIN account_move move ON move.id = line.move_id
+              JOIN account_journal journal ON journal.id = move.journal_id
+              JOIN res_company company ON company.id = journal.company_id
+              JOIN res_currency currency ON currency.id = company.currency_id
+             WHERE line.move_id IN %s
+          GROUP BY line.move_id, currency.decimal_places
+            HAVING ROUND(SUM(line.balance), currency.decimal_places) != 0   
         ''', [tuple(moves.ids)])
 
         query_res = self._cr.fetchall()
         if query_res:
-            error_msg = _("There was a problem with the following move(s):\n")
+            error_msg = _("An error has occured")
             for move in query_res:
-                id_, balance = move
-                error_msg += _("- Move with id %i\n", id_)
-                if balance != 0.0:
-                    error_msg += _("\tCannot create unbalanced journal entry. The balance is equal to %s\n",
-                                   format_amount(self.env, balance, self.env['account.move'].browse(id_).currency_id))
+                move_id, debit, credit = move
+                error_msg += _(", the move is not balanced.\nThe total of debits equals %s and the total of credits equals %s."
+                               "\nYou might want to specify a default account on the miscellaneous journal to automatically balance each move.",
+                               format_amount(self.env, debit, self.env['account.move'].browse(move_id).currency_id),
+                               format_amount(self.env, credit, self.env['account.move'].browse(move_id).currency_id))
```

Comment:
Avoid doing two browse for the same move. It's not a big issue, but still better to avoid.

```suggestion
                move_id, debit, credit = move
                currency = self.env['account.move'].browse(move_id).currency_id
                error_msg += _(", the move is not balanced.\nThe total of debits equals %s and the total of credits equals %s."
                               "\nYou might want to specify a default account on the miscellaneous journal to automatically balance each move.",
                               format_amount(self.env, debit, currency),
                               format_amount(self.env, credit, currency))
```

Reply:
`a default account on the miscellaneous journal to autom....` > 
`a default account on the journal to automa...`

OR specify the journal name, since the default journal is called "miscellaneous operations"